app/models/discussion_reader.rb

```diff
@@ -24,6 +24,10 @@ def set_volume_as_required!
     end
   end
 
+  def participate!
+    update(participating: true) unless self.participating
```

Comment:
Ah, I did want update_attribute here.

Difference is that `update_attribute` isn't subject to validations or before/after hooks (not that we have them here, but traditionally boolean flags like to use update_attribute)